src/dune/context.ml

```diff
@@ -115,23 +114,17 @@ let to_dyn_concise t : Dyn.t = Context_name.to_dyn t.name
 
 let compare a b = Poly.compare a.name b.name
 
-let opam = lazy (Bin.which ~path:(Env.path Env.initial) "opam")
-
-let opam_config_var ~env ~cache var =
-  match Table.find cache var with
-  | Some _ as x -> Fiber.return x
-  | None -> (
-    match Lazy.force opam with
-    | None -> Fiber.return None
-    | Some fn -> (
-      Process.run_capture (Accept Predicate_lang.any) fn ~env
-        [ "config"; "var"; var ]
-      >>| function
-      | Ok s ->
-        let s = String.trim s in
-        Table.set cache var s;
-        Some s
-      | Error _ -> None ) )
+let opam = Memo.Lazy.of_val (Bin.which ~path:(Env.path Env.initial) "opam")
```

Comment:
`Lazy.of_val` is not actually lazy. You want `Lazy.create`.

Reply:
Oh. Done.